web_responsive/static/src/js/web_responsive.js

```diff
@@ -442,26 +445,26 @@ odoo.define('web_responsive', function (require) {
         * The executed action
         */
         _hideMenusByAction: function (action) {
-            var uniq_sel = '[data-action-id='+action.id+']';
+            const uniq_sel = '[data-action-id='+action.id+']';
             // Need close AppDrawer?
-            var menu_apps_dropdown = document.querySelector(
+            const menu_apps_dropdown = document.querySelector(
                 '.o_menu_apps .dropdown');
             $(menu_apps_dropdown).has('.dropdown-menu.show')
                 .has(uniq_sel).find('> a').dropdown('toggle');
             // Need close Sections Menu?
             // TODO: Change to 'hide' in modern Bootstrap >4.1
-            var menu_sections = document.querySelector(
+            const menu_sections = document.querySelector(
                 '.o_menu_sections li.show');
             $(menu_sections).has(uniq_sel).find('.dropdown-toggle')
                 .dropdown('toggle');
             // Need close Mobile?
-            var menu_sections_mobile = document.querySelector(
+            const menu_sections_mobile = document.querySelector(
                 '.o_menu_sections.show');
             $(menu_sections_mobile).has(uniq_sel).collapse('hide');
         },
 
         _handleAction: function (action) {
-            return this._super.apply(this, arguments).always(
+            return this._super.apply(this, arguments).finally(
```

Comment:
AFAIK, if super returns a `Promise` (I think it should now), then the direct replacement for `always()` is `then(fct).guardedCatch(fct)`, where `fct` is the method to be called.

Idea taken from https://github.com/odoo/odoo/pull/31887

Reply:
I think that in this case "finally" (then + catch) it's better than use "then" + "guardedCatch" (catch without errors) because i want to ensure that the menu is hidden in any circumstance. If not, the user can get stucked in the menu. What do you think?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Reply:
Oh definitely, I didn't realize that was the purpose. Thanks for the info :wink: